src/components/common/YourCompany/YourCompany.vue

```diff
@@ -468,32 +470,28 @@ export default class YourCompany extends Mixins(
     return 'Unknown'
   }
 
-  /** Compare names. */
+  /** Whether a new business legal name was entered.. */
   get isNewName () {
+    const originalName = this.getNameRequestLegalName
     const currentName = this.getEntitySnapshot?.businessInfo.legalName
-    return (this.getNameRequestApprovedName !== currentName)
+    return (originalName !== currentName)
   }
 
   /** The current options for change of name correction or edit. */
   get nameChangeOptions (): Array<CorrectionTypes> {
-    // make a copy so we don't change the original resource data
-    const nameChangeOptions = cloneDeep(this.getResource.changeData.nameChangeOptions)
-
-    // Remove name-to-numbered company option when already a numbered company
-    if (!this.getNameRequestNumber) nameChangeOptions.splice(1, 1)
```

Comment:
This sometimes failed when we provided 2 options (it would splice out the wrong item).

The new code works fine in all cases.